yarn-project/p2p/src/service/libp2p_service.ts

```diff
@@ -223,16 +249,52 @@ export class LibP2PService implements P2PService {
       const tx = Tx.fromBuffer(Buffer.from(message.data));
       await this.processTxFromPeer(tx);
     }
+    if (message.topic === BlockAttestation.p2pTopic) {
+      const attestation = BlockAttestation.fromBuffer(Buffer.from(message.data));
+      await this.processAttestationFromPeer(attestation);
+    }
+    if (message.topic == BlockProposal.p2pTopic) {
+      const block = BlockProposal.fromBuffer(Buffer.from(message.data));
+      await this.processBlockFromPeer(block);
+    }
 
     return;
   }
 
+  /**Process Attestation From Peer
+   * When a proposal is received from a peer, we add it to the attestation pool, so it can be accessed by other services.
+   *
+   * @param attestation - The attestation to process.
+   */
+  private async processAttestationFromPeer(attestation: BlockAttestation): Promise<void> {
+    this.logger.verbose(`Received attestation ${attestation.p2pMessageIdentifier()} from external peer.`);
+    await this.attestationPool.addAttestations([attestation]);
```

Comment:
Related to earlier, not sure where the best place to check for it would be, but think we gotta check somewhere to make sure we don't end up thinking that we have enough attestations, but it is because we just stored a lot of garbage attestations.